warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryOptionsFromQueryVisitor.java

```diff
@@ -121,6 +124,65 @@ private Object visitJunction(JexlNode node, Object data, Supplier<JexlNode> crea
+                        UniqueFunction.UNIQUE_BY_MILLISECOND_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_MILLISECOND), UNIQUE_BY_MINUTE(
+                        UniqueFunction.UNIQUE_BY_MINUTE_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_MINUTE), UNIQUE_BY_MONTH(
+                        UniqueFunction.UNIQUE_BY_MONTH_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_MONTH), UNIQUE_BY_SECOND(
+                        UniqueFunction.UNIQUE_BY_SECOND_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_SECOND), UNIQUE_BY_TENTH_OF_HOUR(
+                        UniqueFunction.UNIQUE_BY_TENTH_OF_HOUR_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_TENTH_OF_HOUR), UNIQUE_BY_YEAR(
+                        UniqueFunction.UNIQUE_BY_YEAR_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_YEAR);
+        
+        public String getName() {
+            return name;
+        }
+        
+        public final String name;
+        public final UniqueGranularity function;
+        public static final String UNIQUE_BY_DAY_FUNCTION = "unique_by_day";
+        public static final String UNIQUE_BY_HOUR_FUNCTION = "unique_by_hour";
+        public static final String UNIQUE_BY_MINUTE_FUNCTION = "unique_by_minute";
+        public static final String UNIQUE_BY_TENTH_OF_HOUR_FUNCTION = "unique_by_tenth_of_hour";
+        public static final String UNIQUE_BY_MONTH_FUNCTION = "unique_by_month";
+        public static final String UNIQUE_BY_SECOND_FUNCTION = "unique_by_second";
+        public static final String UNIQUE_BY_MILLISECOND_FUNCTION = "unique_by_millisecond";
+        public static final String UNIQUE_BY_YEAR_FUNCTION = "unique_by_year";
+        
+        UniqueFunction(String name, UniqueGranularity function) {
+            this.name = name;
+            this.function = function;
+        }
+        
+        public static UniqueFunction findByName(String name) {
+            switch (name) {
+                case "unique_by_day":
+                    return UNIQUE_BY_DAY;
+                case "unique_by_hour":
+                    return UNIQUE_BY_HOUR;
+                case "unique_by_minute":
+                    return UNIQUE_BY_MINUTE;
+                case "unique_by_second":
+                    return UNIQUE_BY_SECOND;
+                case "unique_by_millisecond":
+                    return UNIQUE_BY_MILLISECOND;
+                case "unique_by_tenth_of_hour":
+                    return UNIQUE_BY_TENTH_OF_HOUR;
+                case "unique_by_month":
+                    return UNIQUE_BY_MONTH;
+                case "unique_by_year":
+                    return UNIQUE_BY_YEAR;
+                default:
+                    throw new IllegalArgumentException("No " + UniqueFunction.class.getSimpleName() + " exists with the name " + name);
+            }
+        }
+    }
+    
+    private void updateUniqueFields(ASTFunctionNode node, UniqueFields uniqueFields, Map<String,String> optionsMap, UniqueFunction granularity) {
```

Comment:
The parameter "granularity" is really the "uniqueFunction".

Reply:
Good call. Switching the name